beets/ui/__init__.py

```diff
@@ -1159,10 +1159,13 @@ def _open_library(config):
         )
         lib.get_item(0)  # Test database connection.
     except (sqlite3.OperationalError, sqlite3.DatabaseError):
+        message = ""
         log.debug(u'{}', traceback.format_exc())
-        raise UserError(u"database file {0} could not be opened".format(
+        if e.args[0] == "unable to open database file":
+            message = "It might be a permissions problem."
+        raise UserError(u"database file {0} could not be opened.%s".format(
             util.displayable_path(dbpath)
-        ))
+        )%message)
```

Comment:
Oh, cool! Looks like you're almost there. There are actually two ways to go with this:

* Inside mutate, raise a `UserError` and let the top-level (`main`) exception handling print the message.
* Inside `main`, just catch the `OperationalError` directly and print the error message there. (But don't raise a new `UserError`.)

It's actually a difficult call to say what's better, but I like your idea of raising a new, specialized exception in `mutate`. In fact, maybe we can invent a new error exception class, sort of like `InvalidQueryError`, for filesystem access problems like this one?